Constraints/CollectionValidator.php

```diff
@@ -34,7 +34,7 @@ public function validate($value, Constraint $constraint)
         }
 
         if (!\is_array($value) && !($value instanceof \Traversable && $value instanceof \ArrayAccess)) {
-            throw new UnexpectedTypeException($value, 'array or Traversable and ArrayAccess');
+            throw new UnexpectedTypeException($value, 'array, Traversable or ArrayAccess');
```

Comment:
This doesn't look right to me. Just implementing either `Traversable` or `ArrayAccess` (but not both of them) still leads to the exception being thrown.

Reply:
You're right, I dived right in without reading the conditions after the comment in the above issue. How about "Either array, or Traversable and ArrayAccess"?